src/pages/index.js

```diff
@@ -55,41 +137,86 @@ const enableValidation = (config) => {
 
 enableValidation(formValidationOptions);
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardSection.addItems(createCard({ name: data.name, link: data.link }));
-    },
-  },
-  selectors.cardSection
-);
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 function handleImageClick(data) {
   cardPreviewImage.open(data);
 }
 
-cardSection.renderItems();
-
-const newcardPopup = new PopupWithForm(selectors.newCardModal, (cardData) => {
-  cardSection.addItems(createCard({ name: cardData.title, link: cardData.url }));
-  newcardPopup.close();
-  formValidators["card-form"].disableButton();
+//update avatar
+const avatarModal = new PopupWithForm("#modal-avatar", (userData) => {
+  const avatarSubmitButton = document.querySelector(
+    selectors.avatarSubmitButton
+  );
+  avatarSubmitButton.textContent = "Saving... ";
+  avatarModal.setLoading(true);
+  api
+    .updateAvatar(userData.url)
+    .then((res) => {
+      userInfoData.setAvatar(res.avatar);
+      avatarModal.close();
+      formValidators["avatar-form"].disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarModal.setLoading(false);
```

Comment:
` avatarModal.setLoading(false);` is not needed because this method is  **empty**. Please, implement `setLoading` correctly, or delete it